### PR TITLE
[opt](nereids) add join shuffle type in explain shape plan

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes
in previous version, users need to determine join shuffle type by join children mannually. It is error-prone.
we print shuffle type in explain shape plan to improve plan readability
Issue Number: close #xxx

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run feut

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/AbstractPhysicalJoin.java [hunk 1]
```diff
@@ -303,4 +304,20 @@ public boolean isBroadCastJoin() {
         }
         return false;
     }
+
+    protected Join.ShuffleType shuffleType() {
+        if (left() instanceof PhysicalDistribute) {
+            return ShuffleType.shuffle;
+        }
```
[the remark this answers is not in the document]
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/AbstractPhysicalJoin.java:311]: nereids could not generate such plan now

**Comment by englefly** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/AbstractPhysicalJoin.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/AbstractPhysicalJoin.java:311]: after refactor coordinator, we will remove this restirct. so this PR should be considered as we can support this kind of planning and support it

**Comment by englefly** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/AbstractPhysicalJoin.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/AbstractPhysicalJoin.java]: add new enum "shuffleBucket" to denote this type of shuffle

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by englefly** [comment]
run p0

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run feut

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run cloud_p1

**Comment by englefly** [comment]
run cloudp1

**Review by starocean999 — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

**Pull request merged**